### ISSUE TITLE
Remove 'control' on entity but let it be moved by other factors other than its own will?

### ISSUE THREAD

**Issue by arakash92** [collinhover/impactplusplus#84]
So, i have an entity extending ig.Character and it has a stun method.

So when I stun it, I removeControl() on it, then addControl() after a while. But he freezes. He should still fall because of gravity and other factors etc. He should just not be able to do anything on its own.

Is there a way to remove 'movement' control on it only and still let it update?

**Comment by arakash92** [comment]
I found out. (setControllable method). lol :P

**Comment by arakash92** [comment]
Nope. turns out the `setControllable` sets the `controllable` properties which means he can't be moved by other factors while `controllable == false`.

Hm.

**Comment by collinhover** [comment]
Controllable is meant to stop an entity entirely, including ignoring gravity, but is only intended to be used temporarily. There is not yet a built in way to stun an entity. I realize controllable is a misleading name, do you have a good idea for what we can change it to?

To do a stun effect, we just need an extra if statement at https://github.com/collinhover/impactplusplus/blob/master/lib/plusplus/core/entity.js#L4570, maybe checking for controllable, assuming we change the name of controllable to something else.

**Comment by arakash92** [comment]
Hm.. 'frozen' is already used for something else... 'timeFreeze' perhaps?

**Comment by collinhover** [comment]
I'll leave it as is and just add a stunned property instead. No eta on adding this, but it is on my list of additions.

**Comment by arakash92** [comment]
yea, stunned works fine.
